### PR TITLE
109 - Add Celery and separate each part of the run process into endpoints and tasks

### PR DESCRIPTION
**Pull request by ctmartinez1992**
This PR also adds the ability to run the Code Monkey process for a local repo.

### CODE REVIEW

File: labs/celery.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+
+logger = logging.getLogger(__name__)
+
+CELERY_QUEUE_PREFIX = "code-monkey"
+DEFAULT_QUEUE_NAME = CELERY_QUEUE_PREFIX
+LOW_PRIORITY_QUEUE_NAME = f"{CELERY_QUEUE_PREFIX}-low"
+HIGH_PRIORITY_QUEUE_NAME = f"{CELERY_QUEUE_PREFIX}-high"
+
+redis_client = redis.StrictRedis(host="redis", port=6379, db=0, decode_responses=True)
```
Comment (kenvontucky) [inline labs/celery.py:24]: these values must be retrieved from env variables

File: labs/celery.py
Comment (kenvontucky) [inline labs/celery.py:1]: lets split the celery tasks into tasks files, one for github and the other for llm related tasks

File: labs/database/connect.py [hunk 2]
```diff
@@ -15,18 +15,20 @@
 def db_connector():
     def decorator(original_function):
         def new_function(*args, **kwargs):
+            if settings.TEST_ENVIRONMENT:
```
Comment (kenvontucky) [inline labs/database/connect.py:18]: why do we need a specific connection for test environment?

**Review by kenvontucky — CHANGES_REQUESTED** [review]
environemnt sample must be updatd as well to reflect the newly added variables

File: labs/celery.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+
+app = Celery(
+    name="tiktok_connector_worker",
```
[the remark this answers is not in the document]
Comment (ctmartinez1992) [reply labs/celery.py:28]: fixed

File: labs/celery.py (hunk 1 above)
Comment (ctmartinez1992) [reply labs/celery.py]: fixed

File: labs/celery.py
Comment (ctmartinez1992) [reply labs/celery.py:1]: done.

**Comment by ctmartinez1992** [comment]
> environemnt sample must be updatd as well to reflect the newly added variables

This is done in this PR.

File: labs/database/connect.py (hunk 2 above)
Comment (ctmartinez1992) [reply labs/database/connect.py:18]: Added a comment to make it clearer why it's needed.

**Review by JdFSilva — APPROVED** [review]

**Pull request merged**